### PR TITLE
Mathew PR

### PR DESCRIPTION
**Pull request by wabudd1**

### CODE REVIEW

File: Controllers/HomeController.cs [hunk 1]
```diff
@@ -8,9 +8,94 @@ namespace TestWebApp.Controllers
...
+
+        [HttpGet]
+        public ActionResult Display()
+        {
+            //MockDb db = new MockDb();
+
+            //DataRepository dr = new DataRepository();
+
+
+            return this.View(dr);
```
Comment (wabudd1) [inline Controllers/HomeController.cs:75]: `this` keyword not needed

File: Controllers/HomeController.cs [hunk 2]
```diff
@@ -8,9 +8,94 @@ namespace TestWebApp.Controllers
...
+            for (int i = 2; i <= number;i++)  
+            {  
+                Fib[i] = Fib[i - 2] + Fib[i - 1];  
+            }
+            //return Fib[number]; 
+
+            ViewBag.Fibonacci = Fib;
+            //ViewData["Fib"] = Fib;
+
+            return View(ViewBag);
```
Comment (wabudd1) [inline Controllers/HomeController.cs:98]: `ViewBag` isn't really used much.  I would recommend creating a ViewModel class instead.

File: Controllers/HomeController.cs [hunk 3]
```diff
@@ -8,9 +8,94 @@ namespace TestWebApp.Controllers
 {
     public class HomeController : Controller
     {
+               DataRepository dr = new DataRepository();
+        MockDb db = new MockDb();
```
Comment (wabudd1) [inline Controllers/HomeController.cs:12]: Don't instantiate these as globals.  Use the repository only when needed within a method call.

File: Controllers/HomeController.cs [hunk 4]
```diff
@@ -8,9 +8,94 @@ namespace TestWebApp.Controllers
...
+
+
+            
+
+        }
+
+        [HttpGet]
+        public ActionResult Display()
+        {
+            //MockDb db = new MockDb();
```
Comment (wabudd1) [inline Controllers/HomeController.cs:70]: It looks like you're trying to return the entire repository to the View, which is less than ideal.  Usually you want to have strong separation between your client side and server side code.  For this, I usually like to call BLL methods from the controller.  From a BLL method, I would call a repository method, which in turn calls the database.

The goal for this method should be to obtain a collection of `User` objects from *somewhere* and return them as part of a ViewModel with the View.

File: Data/DataRepository.cs [hunk 5]
```diff
@@ -19,9 +19,17 @@ public static List<User> GetAllUsers()
...
+            //throw new NotImplementedException();
+
+            using (MockDb db = new MockDb())
+            {
+                db.AddUser(user);
+
+            }
+
+
         }
```
Comment (wabudd1) [inline Data/DataRepository.cs:33]: This is a technically correct implementation.  For simplicity sake, a `User` for the parameter is good.  In the real world, you would use something else and then later on convert it before it goes to the database.

I would not call the DB directly in a controller method.  A repository method or, ideally, a BLL Method would be better.

File: Controllers/HomeController.cs (hunk 3 above)
Comment (nachochips18) [reply Controllers/HomeController.cs:12]: so those two lines are unnecessary?

Comment (wabudd1) [reply Controllers/HomeController.cs:12]: Correct.  I would delete them.

**Comment by nachochips18** [comment]
Is there a reason Model is an empty folder?

**Comment by wabudd1** [comment]
> Is there a reason Model is an empty folder?

No reason.  Just how Visual Studio sets up new .NET MVC projects.  I just decided not to put anything there.

**Pull request closed without merging**